### PR TITLE
rename sweep to redeem_onchain_funds

### PR DESCRIPTION
**Pull request by ubbabeck**
In order to avoid confusion about sweep which implies sending funds from the closing address to the internal wallet. By renaming it to withdraw we indicate that it is sending the funds out.

### CODE REVIEW

**Comment by JssDWt** [comment]
I find `withdraw` more confusing than `sweep` tbh. We have lnurl withdraw for example, which means receiving funds into your sdk balance. Withdrawing something is usually incoming. The terms deposit/withdraw are generally confusing and should be avoided altogether imo. Send and receive are clearer, because the direction is clear.

Sweep means claim onchain funds that are already yours. Since you cannot access your onchain funds through the sdk, and you have to provide an address, the ring for me is "claim funds that are already yours into this specific address".

**Comment by kingonly** [comment]
> I find `withdraw` more confusing than `sweep` tbh. We have lnurl withdraw for example, which means receiving funds into your sdk balance. Withdrawing something is usually incoming. The terms deposit/withdraw are generally confusing and should be avoided altogether imo. Send and receive are clearer, because the direction is clear.
> 
> Sweep means claim onchain funds that are already yours. Since you cannot access your onchain funds through the sdk, and you have to provide an address, the ring for me is "claim funds that are already yours into this specific address".

The problem is that "sweep" has a specific semantics in Lightning (moved funds from a closed channel to the node's wallet). Our API is different, it's about moving funds from the node's wallet to another BTC address. That's why "withdraw on-chain" or something would be better imo.

**Comment by kingonly** [comment]
"transfer on chain funds"?

**Comment by dangeross** [comment]
I agree with @JssDWt, withdraw can be confusing because it used elsewhere as an incoming/receiving action. I think `transfer_onchain` could work like @kingonly said

**Comment by kingonly** [comment]
Not sure we can do better with transfer. I agree withdraw isn't great, but sweep is very confusing to Lightning devs and I don't think we should overload it with new semantics. Open to suggestions.

**Comment by ubbabeck** [comment]
I agree that withdraw also can cause confusion since this is a lighting first sdk. One suggestion is to use transaction as it is reserved for on-chain and payment is reserved for lighting. 

Suggestions  `sweepTransaction` or `withdrawTransaction`. Thought they might be a bit too verbose when combined eg.  `PrepareWithdrawTransactionReqeust` or `PrepareSweepTransactionRequest`.

**Comment by kingonly** [comment]
It doesn't help. Sweep transaction is even worse because it refers to a very specific transaction (moving the funds from a closed channel to an the node's on-chain wallet). transfer_onchain_funds (which echo the 'unexpected funds' UI is the best option so far imo.

**Comment by ubbabeck** [comment]
Since the last commit I've renamed the sweep to transfer_onchain_funds and prepare_sweep to prepare_transfer_onchain_funds. 

Do we also wish to use the same type of unsigned integer for both of the methods? 

Which of u32 or u64 is preferred?

**Comment by dangeross** [comment]
u32 is the most commonly used for sat_per_vbyte.

My other options are `remit_onchain` or `forward_onchain`

**Comment by ok300** [comment]
To indicate we're dealing with onchain, we could include `utxo` in the name, like `sweep_utxos` or `transfer_utxos`.

One step further, to include directionality: `transfer_utxos_to` (although IMO `sweep_utoxs` includes direction and separates it from LN semantics).

**Comment by kingonly** [comment]
on-chain is better and clearer than utxo.
transfer_onchain is still better the remit or forward.

**Review by roeierez — APPROVED** [review]
LGTM

File: libs/sdk-bindings/src/uniffi_binding.rs [hunk 1]
```diff
@@ -195,7 +196,7 @@ impl BlockingBreezServices {
         rt().block_on(self.breez_services.report_issue(req))
     }
 
-    pub fn sweep(&self, req: SweepRequest) -> SdkResult<SweepResponse> {
+    pub fn sweep(&self, req: RedeemOnchainFundsRequest) -> SdkResult<RedeemOnchainFundsResponse> {
         rt().block_on(self.breez_services.sweep(req))
     }
```
Comment (dangeross) [inline libs/sdk-bindings/src/uniffi_binding.rs:201]:
```suggestion
    pub fn redeem_onchain_funds(&self, req: RedeemOnchainFundsRequest) -> SdkResult<RedeemOnchainFundsResponse> {
        rt().block_on(self.breez_services.redeem_onchain_funds(req))
    }
```

File: libs/sdk-core/src/binding.rs [hunk 2]
```diff
@@ -360,15 +361,22 @@ pub fn buy_bitcoin(req: BuyBitcoinRequest) -> Result<BuyBitcoinResponse> {
 }
 
 /// See [BreezServices::sweep]
```
Comment (dangeross) [inline libs/sdk-core/src/binding.rs:363]:
```suggestion
/// See [BreezServices::redeem_onchain_funds]
```

File: libs/sdk-core/src/binding.rs [hunk 3]
```diff
@@ -360,15 +361,22 @@ pub fn buy_bitcoin(req: BuyBitcoinRequest) -> Result<BuyBitcoinResponse> {
...
-pub fn sweep(req: SweepRequest) -> Result<SweepResponse> {
+pub fn sweep(req: RedeemOnchainFundsRequest) -> Result<RedeemOnchainFundsResponse> {
     block_on(async { get_breez_services().await?.sweep(req).await })
         .map_err(anyhow::Error::new::<SdkError>)
 }
```
Comment (dangeross) [inline libs/sdk-core/src/binding.rs:367]:
```suggestion
pub fn redeem_onchain_funds(req: RedeemOnchainFundsRequest) -> Result<RedeemOnchainFundsResponse> {
    block_on(async { get_breez_services().await?.redeem_onchain_funds(req).await })
        .map_err(anyhow::Error::new::<SdkError>)
}
```

File: libs/sdk-core/src/breez_services.rs [hunk 4]
```diff
@@ -576,20 +576,26 @@ impl BreezServices {
         Ok(self.persister.get_payment_by_hash(&hash)?)
     }
 
-    /// Sweep on-chain funds to the specified on-chain address, with the given feerate
-    pub async fn sweep(&self, req: SweepRequest) -> SdkResult<SweepResponse> {
+    /// RedeemOnchainFunds on-chain funds to the specified on-chain address, with the given feerate
+    pub async fn sweep(
```
Comment (dangeross) [inline libs/sdk-core/src/breez_services.rs:580]:
```suggestion
    pub async fn redeem_onchain_funds(
```

File: libs/sdk-core/src/node_api.rs [hunk 5]
```diff
@@ -90,7 +90,7 @@ pub trait NodeAPI: Send + Sync {
         last_hop: Option<&RouteHintHop>,
     ) -> NodeResult<Vec<MaxChannelAmount>>;
     async fn sweep(&self, to_address: String, sat_per_vbyte: u32) -> NodeResult<Vec<u8>>;
```
Comment (dangeross) [inline libs/sdk-core/src/node_api.rs:92]:
```suggestion
    async fn redeem_onchain_funds(&self, to_address: String, sat_per_vbyte: u32) -> NodeResult<Vec<u8>>;
```

File: libs/sdk-bindings/src/breez_sdk.udl [hunk 6]
```diff
@@ -777,7 +777,7 @@ interface BlockingBreezServices {
    sequence<Payment> list_payments(ListPaymentsRequest req);
 
    [Throws=SdkError]
-   SweepResponse sweep(SweepRequest req);
+   RedeemOnchainFundsResponse sweep(RedeemOnchainFundsRequest req);
```
Comment (dangeross) [inline libs/sdk-bindings/src/breez_sdk.udl:780]:
```suggestion
   RedeemOnchainFundsResponse redeem_onchain_funds(RedeemOnchainFundsRequest req);
```

File: libs/sdk-core/src/breez_services.rs [hunk 7]
```diff
@@ -576,20 +576,26 @@ impl BreezServices {
         Ok(self.persister.get_payment_by_hash(&hash)?)
     }
 
-    /// Sweep on-chain funds to the specified on-chain address, with the given feerate
-    pub async fn sweep(&self, req: SweepRequest) -> SdkResult<SweepResponse> {
+    /// RedeemOnchainFunds on-chain funds to the specified on-chain address, with the given feerate
```
Comment (dangeross) [inline libs/sdk-core/src/breez_services.rs:579]:
```suggestion
    /// Redeem on-chain funds from closed channels to the specified on-chain address, with the given feerate
```

**Review by dangeross — CHANGES_REQUESTED** [review]
Seems to be a couple of places where functions are still named `sweep`.

File: libs/sdk-bindings/src/uniffi_binding.rs (hunk 1 above)
Comment (ubbabeck) [reply libs/sdk-bindings/src/uniffi_binding.rs]: Oh, my bad. Thanks, fixed with the latest push.

File: libs/sdk-core/src/binding.rs (hunk 3 above)
Comment (ubbabeck) [reply libs/sdk-core/src/binding.rs]: Done

File: libs/sdk-core/src/binding.rs (hunk 2 above)
Comment (ubbabeck) [reply libs/sdk-core/src/binding.rs]: Done

File: libs/sdk-core/src/breez_services.rs (hunk 7 above)
Comment (ubbabeck) [reply libs/sdk-core/src/breez_services.rs]: Updated

File: libs/sdk-core/src/models.rs [hunk 8]
```diff
@@ -1298,21 +1298,21 @@ pub enum BuyBitcoinProvider {
     Moonpay,
 }
 
-/// We need to prepare a sweep transaction to know what fee will be charged in satoshis this
-/// model holds the request data which consists of the address to sweep to and the fee rate in
+/// We need to prepare a redeem_onchain_funds transaction to know what fee will be charged in satoshis this
+/// model holds the request data which consists of the address to redeem_onchain_funds to and the fee rate in
```
Comment (dangeross) [inline libs/sdk-core/src/models.rs:1302]:
```suggestion
/// We need to prepare a redeem_onchain_funds transaction to know what fee will be charged in satoshis. 
/// This model holds the request data which consists of the address to redeem on-chain funds to and the fee rate in
```

File: tools/sdk-cli/src/command_handlers.rs [hunk 9]
```diff
@@ -232,29 +232,29 @@ pub(crate) async fn handle_command(
...
             sat_per_vbyte,
         } => {
             sdk()?
-                .sweep(SweepRequest {
+                .redeem_onchain_funds(RedeemOnchainFundsRequest {
                     to_address,
                     sat_per_vbyte,
                 })
                 .await?;
             Ok("Onchain funds were swept successfully".to_string())
```
Comment (dangeross) [inline tools/sdk-cli/src/command_handlers.rs:245]: Maybe change this so we can see the txid
```suggestion
            serde_json::to_string_pretty(&resp).map_err(|e| e.into())
```

File: libs/sdk-core/src/models.rs [hunk 10]
```diff
@@ -1298,21 +1298,21 @@ pub enum BuyBitcoinProvider {
...
-/// We need to prepare a sweep transaction to know what a fee it will be charged in satoshis
+/// We need to prepare a redeem_onchain_funds transaction to know what a fee it will be charged in satoshis
 /// this model holds the response data, which consists of the weight and the absolute fee in sats
 #[derive(PartialEq, Eq, Debug, Clone, Deserialize, Serialize)]
-pub struct PrepareSweepResponse {
-    pub sweep_tx_weight: u64,
-    pub sweep_tx_fee_sat: u64,
+pub struct PrepareRedeemOnchainFundsResponse {
+    pub redeem_onchain_funds_tx_weight: u64,
+    pub redeem_onchain_funds_tx_fee_sat: u64,
```
Comment (dangeross) [inline libs/sdk-core/src/models.rs:1315]: Same renaming as above

**Review by dangeross — CHANGES_REQUESTED** [review]
Looking good. Just a couple more bits I missed, sorry 🙂

File: libs/sdk-bindings/src/breez_sdk.udl [hunk 11]
```diff
@@ -656,22 +656,22 @@ enum BuyBitcoinProvider {
...
-    u64 sat_per_vbyte;
+    u32 sat_per_vbyte;
 };
 
-dictionary PrepareSweepResponse {
-    u64 sweep_tx_weight;
-    u64 sweep_tx_fee_sat;
+dictionary PrepareRedeemOnchainFundsResponse {
+    u64 redeem_onchain_funds_tx_weight;
+    u64 redeem_onchain_funds_tx_fee_sat;
```
[the remark this answers is not in the document]
Comment (ubbabeck) [reply libs/sdk-bindings/src/breez_sdk.udl]: Thanks, much more digestible.

File: libs/sdk-core/src/models.rs (hunk 8 above)
Comment (ubbabeck) [reply libs/sdk-core/src/models.rs]: Done

File: tools/sdk-cli/src/command_handlers.rs (hunk 9 above)
Comment (ubbabeck) [reply tools/sdk-cli/src/command_handlers.rs]: Added

File: libs/sdk-core/src/models.rs (hunk 10 above)
Comment (ubbabeck) [reply libs/sdk-core/src/models.rs]: Done

**Review by dangeross — APPROVED** [review]
LGTM! 🚀

**Pull request merged**